tests/jobs/scala/src/main/scala/KerberizedKafkaProducer.scala

```diff
@@ -33,16 +37,63 @@ object KerberizedKafkaProducer {
     producer.close()
   }
 
-  def sendMessages(topic: String, messages: Array[String])(producer: KafkaProducer[Int, String]): Unit = {
-    messages.foreach { str =>
-      val msg = new ProducerRecord[Int, String](topic, str.hashCode, str)
-      println(s"sending message: $msg")
+  def sendMessages(topic: String, messages: List[String])(producer: KafkaProducer[Int, String]): Unit = {
+    import ExecutionContext.Implicits.global
 
-      Try(producer.send(msg)) match {
-        case Success(_) => println("Message sent")
-        case Failure(ex) => ex.printStackTrace()
+    val messageCounter = new AtomicInteger()
+
+    val futures = messages.map { str =>
+      val msg = new ProducerRecord[Int, String](topic, str)
+      sendMessage(producer, msg)
+    }
+
+    futures.foreach { future =>
+      future.onComplete {
+        case Success(metadata) =>
+          println(s"Message has been published to (topic_partition@offset): ${metadata.toString}")
+          messageCounter.incrementAndGet()
+        case Failure(exception) => exception.printStackTrace()
       }
     }
+
+    val resultFuture = Future.sequence(futures)
+    Await.result(resultFuture, timeout)
+
+    println(
+      s"""
+         |${messageCounter.get()} messages sent to Kafka
+         |===========================================================================
+         |Producer metrics by topic:
+       """.stripMargin)
+
+    producer.metrics.values
+      .filter(m => m.metricName().group().equalsIgnoreCase("producer-topic-metrics"))
+      .foreach(metric => println(metric.metricName().description() + ": " + metric.metricValue()))
+
+    println(
+      s"""
+        |===========================================================================
+        |Partition metadata for topic '$topic':
+        |${producer.partitionsFor(topic).mkString("\n")}
+      """.stripMargin)
```

Comment:
Looks like you pulled in some unintended commits. Please rebase with `master` so that you only have your changes in the PR.

Reply:
Ok, Sam. This unintended commit came after rebasing with master, I checked this change is present in the Master.
This PR was looking messy, created a new fresh PR -> https://github.com/mesosphere/spark-build/pull/522 
Please have a look.